src/core_plugins/kibana/public/discover/__tests__/directives/field_calculator.js

```diff
@@ -41,6 +41,15 @@ describe('fieldCalculator', function () {
         .to.throwError();
     });
 
+    it('should handle values with dots in them', function () {
+      values = ['0', '0.........', '0.......,.....'];
+      params = {};
+      groups = fieldCalculator._groupValues(values, params);
+      expect(groups['0'].count).to.be(1);
+      expect(groups['0.........'].count).to.be(1);
+      expect(groups['0.......,.....'].count).to.be(1);
```

Comment:
Could you change these three lines and reference `values` directly? Like `expect(groups[values[0]].count).to.be(1);`